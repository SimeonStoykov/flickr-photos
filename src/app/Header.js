import React from 'react';

const Header = () => {
	return (
		<header className="header">
			<h1 className="header--title">Flickr Photos</h1>
			<h2>Other text</h2>
		</header>
	);
}

export default Header;
