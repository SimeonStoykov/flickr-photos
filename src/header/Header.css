.header {
  background-color: #1f1f22;
  color: #fff;
  position: fixed;
  top: 0;
  width: 100%;
  height: 60px;
  line-height: 56px;
}

.header--title {
  cursor: pointer;
  display: inline-block;
  margin: 0 5vw;
  font-size: calc(15px + 1vw);
  vertical-align: middle;
  height: 100%;
}

.header--search-bar {
  margin: 0 auto;
  display: inline-block;
  height: 100%;
}

.header--search-bar input {
  width: 220px;
  height: 26px;
  padding-left: 5px;
  background-color: #d2d2d3;
  outline: 0;
  border: 0;
  border-radius: 2px;
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
  transition: background-color .25s ease-in-out,left .25s ease-in-out,width .25s ease-in-out,opacity .25s ease-in-out;
}

.header--search-bar button {
  height: 28px;
  border: 0;
  outline: 0;
  border-radius: 2px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  cursor: pointer;
  transition: background-color .25s ease-in-out,left .25s ease-in-out,width .25s ease-in-out,opacity .25s ease-in-out;
  background: url('../images/magnifier.svg');
  background-position: center;
  background-size: 20px;
  background-repeat: no-repeat;
  background-color: #d2d2d3;
  width: 35px;
  vertical-align: middle;
  padding: 0;
}

.header--search-bar input:focus,
.header--search-bar input:focus + button {
  background-color: #fff;
}

.header--search-bar button:disabled {
  background: url('../images/magnifier_disabled.svg');
  background-position: center;
  background-size: 20px;
  background-repeat: no-repeat;
  background-color: #d2d2d3;
  cursor: not-allowed;
}

.header--search-bar input:focus {
  width: 300px;
}

::placeholder { /* Chrome, Firefox, Opera, Safari 10.1+ */
  color: #848485;
  opacity: 1; /* Firefox */
}

:-ms-input-placeholder { /* Internet Explorer 10-11 */
  color: #848485;
}

::-ms-input-placeholder { /* Microsoft Edge */
  color: #848485;
}

/* under 570 px make search bar drop down on another row */