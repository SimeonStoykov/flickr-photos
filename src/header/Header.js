import React from 'react';
import PhotoSearchBar from '../photo/PhotoSearchBar';
import history from '../app/history';
import './Header.css';

const HOME_PATH = '/';

const Header = ({ currentPath }) => {
	function loadHomePage() {
		if (currentPath !== HOME_PATH) {
			history.push(HOME_PATH);
		} else {
			window.location.reload();
		}
	}

	return (
		<header className="header">
			<h1 className="header--title" onClick={loadHomePage}>Flickr Photos</h1>
			<PhotoSearchBar />
		</header>
	);
}

export default Header;
