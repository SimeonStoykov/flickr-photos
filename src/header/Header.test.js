import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import '../setupTests';

describe('Header', () => {
    it('should render correctly with no props', () => {
        const component = shallow(<Header />);

        expect(component).toMatchSnapshot();
    });

    it('should render banner text correctly with given currentPath', () => {
        const currentPath = '/';
        const component = shallow(<Header currentPath={currentPath} />);
        expect(component).toMatchSnapshot();
    });
});