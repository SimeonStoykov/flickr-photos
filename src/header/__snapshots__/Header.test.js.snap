// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render banner text correctly with given currentPath 1`] = `
<header
  className="header"
>
  <h1
    className="header--title"
    onClick={[Function]}
  >
    Flickr Photos
  </h1>
  <Connect(PhotoSearchBar) />
</header>
`;

exports[`Header should render correctly with no props 1`] = `
<header
  className="header"
>
  <h1
    className="header--title"
    onClick={[Function]}
  >
    Flickr Photos
  </h1>
  <Connect(PhotoSearchBar) />
</header>
`;
